### PR TITLE
Provide value when assertThatThrownBy/thenThrownBy fail

### PR DESCRIPTION
**Pull request by findepi**
#### Check List:
* Fixes https://github.com/assertj/assertj/issues/3043
* Unit tests : YES
* Javadoc with a code example (on API only) : NO, not yet
* PR meets the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md) -- i hope so!

Following the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md) will make it easier for us to review and accept your PR.

### CODE REVIEW

File: assertj-core/src/main/java/org/assertj/core/api/Assertions.java [hunk 1]
```diff
@@ -1210,6 +1212,17 @@ public static <T extends Throwable> AbstractThrowableAssert<?, T> assertThat(T a
...
+  @CanIgnoreReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assertThatThrownBy(ThrowingCallableWithValue shouldRaiseThrowable) {
+    Object value;
+    try {
+      value = shouldRaiseThrowable.call();
+    } catch (Throwable throwable) {
+      return assertThat(throwable);
+    }
+    throw Failures.instance().failure(String.format("Expecting code to raise a throwable, but it returned [%s] instead", value));
+  }
```
Comment (findepi) [inline assertj-core/src/main/java/org/assertj/core/api/Assertions.java:1224]: I am pretty sure that this is not how it should eventually look like.
Very minimum would be a javadoc.
This is for issue illustration / POC at this stage.

Happy to improve if this is at least directionally correct.

**Comment by joel-costigliola** [comment]
Thanks @findepi, I will review your PR shortly!

File: assertj-core/src/main/java/org/assertj/core/api/Assertions.java (hunk 1 above)
Comment (joel-costigliola) [reply assertj-core/src/main/java/org/assertj/core/api/Assertions.java:1225]: This looks good to me.

We need to do the same in BDDAssertions

**Review by joel-costigliola — CHANGES_REQUESTED** [review]

**Comment by findepi** [comment]
Thanks @joel-costigliola for your review!
added similar change in BDDAssertions. Also added this for overloads that take `String description, Object... args`.

**Comment by joel-costigliola** [comment]
Integrated thanks @findepi!

**Comment by findepi** [comment]
thank you Joel!

**Pull request closed without merging**